### ISSUE TITLE
[suggestion] Tutorial/Handbook addition repeadedcv

### ISSUE THREAD

**Issue by andzandz11** [topepo/caret#685]
Can you please clarify whether it makes sense or not to use nested cv? 

He is confused, and I am too:
https://stats.stackexchange.com/questions/189364/caret-repeated-k-fold-cross-validation-vs-nested-k-fold-cross-validation-repe

You say no:
https://stats.stackexchange.com/questions/125843/outer-crossvalidation-cycle-in-caret-package-r

He says it's an alternative to split your data into test and training, which would be a HUGE benefit (comment):
https://stats.stackexchange.com/questions/225949/do-we-need-a-test-set-when-using-k-fold-cross-validation

He says you don't need test and training dataset at all and just can go along with cross validation:
https://stats.stackexchange.com/questions/148688/cross-validation-with-test-data-set
He agrees:
https://stats.stackexchange.com/questions/261643/do-i-need-a-global-test-set-when-using-k-fold-cross-validation-and-a-small-n-n?noredirect=1&lq=1

He says to never use your test data for model parameters:
https://stats.stackexchange.com/questions/152907/how-do-you-use-test-data-set-after-cross-validation

Much confusion going on!

I think just adding one or two lines in the documentary would elimate a lot of these questions. Something like:
If your dataset iss too small using (nested?) repeatedcv will deliver sufficient power to make your model statistically sound and not a balloon filled with hot air. You lose power by not splitting into a testing and training set, but it is okay.
or:
Ever ever ever use a training and testing set, or you can flush your model down the toilet.

**Comment by topepo** [comment]
First, there are no uniformly correct answers to these questions. Good answers will be very situationally dependent. 

The test set/no test set issue is an old one regardless of how the resampling is done. My thoughts are:

 1. yes resampling methods are very effective and holdout data for only testing is an inefficient use of the data; everything should go into estimating model parameters. This is especially true in small sample sizes.
 1. however... resampling can be done poorly by leaving out other aspects of the model building process (like preprocess, feature selection, subsampling, etc). You wouldn't know if you messed that up without a test set (even a small one). 

I tend to agree with the latter point. It is simply good science to do this, even when it hurts your training set. It's like leaving positive or negative controls out of an experiment. The gain in efficiency that you get is paid for by the risk that you did things wrong and won't know it was wrong until much later. I have paid this price before. 

The main benefit of nested resampling is that it helps mitigate _optimization bias_. Since we choose the best results based on resampling, we might be poorly estimating the true performance. In other words, we are acting like there is no noise in the model tuning process. That is absolutely true but I've never seen a case in my own work where that bias was large in comparison to the other sources of noise. Nested resampling has a huge computational cost and, I don't think, is always worth the benefit. 

It won't be in `caret` for (quite) a while, but I just sent a package to CRAN that has infrastructure to make nested resampling easier to implement called [`rsample`](https://topepo.github.io/rsample/). 

> I think just adding one or two lines in the documentary would eliminate a lot of these questions. 

It might, but documenting good principals in a software manual isn't really a good place for that. If we did that a lot, the documentation becomes too big for people to go through (and it might already be there). SO is a somewhat better location for the discussion. Besides, I don't think that this is an issue that I or anyone else could make a definitive recommendation on. I can only describe my experiences.
